gateway/src/cluster/builder.rs

```diff
@@ -9,14 +9,31 @@ use crate::{
     shard::{tls::TlsContainer, LargeThresholdError, ResumeSession, ShardBuilder},
     EventTypeFlags,
 };
-use std::{collections::HashMap, sync::Arc};
+use std::{
+    collections::HashMap,
+    fmt::{Debug, Formatter, Result as FmtResult},
+    sync::Arc,
+};
 use twilight_gateway_queue::{LocalQueue, Queue};
 use twilight_http::Client;
 use twilight_model::gateway::{
     payload::outgoing::{identify::IdentifyProperties, update_presence::UpdatePresencePayload},
     Intents,
 };
 
+/// Trait helper for [`ShardPresence`] that allows implementing [`Debug`].
+pub trait ShardPresenceTrait: Fn(u64) -> Option<UpdatePresencePayload> + Send + Sync {}
+
+/// Function used in [`ClusterBuilder`] to give shards custom presences on
+/// identify.
+pub type ShardPresence = Box<dyn ShardPresenceTrait<Output = Option<UpdatePresencePayload>>>;
```

Comment:
Add these to the `pub use self::{ .. }` in `mod.rs`

Reply:
I don't think that's necessary because users probably won't ever use the type in their own code, they will just call the function on the builder

Reply:
The problem is that the `shard_presence` method takes a `ShardPresence`. the method should internally box and convert to dyn trait imo

Reply:
right, okay. let me figure that one out